staging/src/k8s.io/apimachinery/pkg/api/meta/multirestmapper.go

```diff
@@ -179,7 +179,7 @@ func (m MultiRESTMapper) RESTMapping(gk schema.GroupKind, versions ...string) (*
 	if len(errors) > 0 {
 		return nil, utilerrors.NewAggregate(errors)
 	}
-	return nil, &NoKindMatchError{PartialKind: gk.WithVersion("")}
+	return nil, &NoKindMatchError{PartialKind: gk.WithVersion(strings.Join(versions, "&"))}
```

Comment:
joining multiple versions into a single string field like this isn't appropriate

Reply:
we can't assume this is only used for display... PartialKind should not contain inaccurate information... if we want to include additional information for display (like all the versions we tried), let's do that in a well-structured way

Reply:
for example, this would improve the way we present the error while preserving well-structured data in the error:

```go
// NoKindMatchError is returned if the RESTMapper can't find any match for a kind
type NoKindMatchError struct {
	// GroupKind is the API group and kind that was searched
	GroupKind schema.GroupKind
	// SearchedVersions is the optional list of versions the search was restricted to
	SearchedVersions []string
}

func (e *NoKindMatchError) Error() string {
	searchedVersions := sets.NewString()
	for _, v := range e.SearchedVersions {
		searchedVersions.Insert(schema.GroupVersion{Group: e.GroupKind.Group, Version: v}.String())
	}

	switch len(searchedVersions) {
	case 0:
		return fmt.Sprintf("no matches for kind %q in group %q", e.GroupKind.Kind, e.GroupKind.Group)
	case 1:
		return fmt.Sprintf("no matches for kind %q in version %q", e.GroupKind.Kind, searchedVersions.List()[0])
	default:
		return fmt.Sprintf("no matches for kind %q in versions %q", e.GroupKind.Kind, searchedVersions.List())
	}
}
```